include/Util/Input.hpp

```diff
@@ -14,36 +14,38 @@ namespace Util {
 
 class Input {
 public:
+    static std::shared_ptr<Input> GetInstance();
+
     std::pair<int, int> GetScrollDistance() const;
-    std::pair<int, int> GetCursorPosition() const;
 
+    std::pair<int, int> GetCursorPosition() const;
     bool IsKeyPressed(const Keycode &key) const;
     bool IsLButtonPressed() const;
     bool IsRButtonPressed() const;
     bool IsMButtonPressed() const;
     bool IfScroll() const;
+
     bool IsMouseMoving() const;
 
     void Update();
 
-    static std::shared_ptr<Input> GetInstance();
-
 private:
     Input() = default;
 
+    static std::shared_ptr<Input> s_Instance;
+
     SDL_Event m_Event;
 
     const Uint8 *m_KeyState = SDL_GetKeyboardState(nullptr);
     std::pair<int, int> m_CursorPosition = {-1, -1};
-    std::pair<int, int> m_ScrollDistance = {0, 0};
 
+    std::pair<int, int> m_ScrollDistance = {0, 0};
```

Comment:
we can change this to use `glm::vec2` for better consistency now

Reply:
https://github.com/ntut-open-source-club/practical-tools-for-simple-design/commit/f803673791e68fa4cc9b5428af12bc4e4bc02a78